templates/default/bump_environments.rb.erb

```diff
@@ -44,8 +51,16 @@ Git.configure do |config|
 end
 git = Git.open('.')
 git.branch('master').checkout
-chef_envs_string = all_envs ? 'all-envs' : chef_envs.join(',')
-git_branch = "jenkins/bump-#{cookbook}-to-#{version}-in-#{chef_envs_string}"
+branch_hash = if is_all_envs
+                'in-all-envs'
+              elsif is_default_envs
+                'in-default-envs'
+              else
+                ''
+              end
+# Add a 10-digit hash of the version and envs so the branch names are unique.
+branch_hash += "-#{(chef_envs.to_a << version).join(',').hash % 10_000_000_000}"
```

Comment:
If you are creating a hash why not just grab the first 10 characters of the hash? why the mod operation?